storage/innobase/pars/pars0pars.cc

```diff
@@ -97,7 +97,13 @@ ib_mutex_t pars_mutex;
 void pars_init() { mutex_create(LATCH_ID_PARSER, &pars_mutex); }
 
 /** Clean up the internal parser */
-void pars_close() { mutex_free(&pars_mutex); }
+void pars_close() {
+  /* pars_init() is called after buf_pool_init(), and if buf_pool_ptr !=
+   * nullptr, then pars_mutex was created */
+  if (!buf_pool_ptr) return;
```

Comment:
Same as above, Its better to use a internal parser state (or a corresponding global variable) instead of buf_pool